wordpress-mobile/WordPress-Android
# Category Management: Add category implementation

Fixes 
#16072 

This PR adds a screen for creating a new category.

<img src="https://user-images.githubusercontent.com/17463767/158403523-dde83b4c-d83a-4471-8fd0-9b6a67654ea3.png" width =400/>

## To test:

Prerequisites
- Go to My Site -> Me -> App Settings -> Test Feature Configuration
- Enable ManageCategoriesFeatureConfig and restart the app
- Relaunch the app.

### Test 1
- Go to Categories Screen
- Press FAB button to launch Add new Category 
- Notice that by Default `Top level` Category is selected in Parent Category Spinner
- Notice that only on Typing the category name, the`Add Category` button gets enabled 
- On Clicking `Add Category` button, progress dialog is shown 
- Notice that On Success, Toast with message `Category added successfully` is shown 
- Notice that the User is redirected to the Categories Listing screen with the Added category present 


### Test 2 - Adding Category Failed 
- Follow the steps as in Test 1 and create a Category 
- Try to create a duplicate category with same name and same level
- Notice that a Toast with message `Adding category failed` is shown 



## Regression Notes
1. Potential unintended areas of impact
Categories Listing screen not being shown properly when new category is added 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests 

3. What automated tests I added (or what prevented me from doing so)
None 

PR submission checklist:
